client/knowledge_base_search/src/containers/ResultsContainer/ResultComparison.js

```diff
@@ -32,49 +33,80 @@ class ResultComparison extends Component {
   }
 
   getFullAnswer() {
-    const { full_result_type, result, enriched_result } = this.props;
+    const { full_result_type, result, enriched_result, passages } = this.props;
 
-    return full_result_type === 'enriched' ? enriched_result : result;
+    switch(full_result_type) {
+      case 'enriched':
+        this.setState({full_answer: enriched_result});
+        break;
+      case 'passages':
+        Promise.all([
+          query('enriched', {query: "id: "+passages.document_id})
+        ]).then((doc_by_id) => {
```

Comment:
they are already cached by the browser by default with this method. we can explore later if the better option is to query ahead of time---I think it's just better to query individually for now so that the user experience isn't ruined at the top level if a query for one of the documents fails